specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/stable/2025-06-01/diagnostic.json

```diff
@@ -0,0 +1,307 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "ComputeDiagnosticResourceProviderClient",
+    "description": "The Compute Diagnostic Resource Provider Client",
+    "version": "2025-06-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/placementScores/spot": {
```

Comment:
>/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/placementScores/spot [](http://example.com/codeflow?start=5&length=101)

This is a singleton instance of resourceType locations/placementScores. Was that intentional? If so, it's normally suggested to use the name "default" instead of "spot". You will also need to add a collection GET path for the locations/placementScores type (will just return a list with one item in it).

If something else is intended, this API may need to be changed.

Reply:
Same as above - adding new API version to an already existing API that has been rolled out in private versions


There is an internal placementScores/general API which will also be exposed via ARM in future versions